datastore/kafka/kafka.go

```diff
@@ -74,21 +72,17 @@ func (p *Producer) Produce(entry *telemetry.Record) {
 		Key:            []byte(entry.Vin),
 		Headers:        headersFromRecord(entry),
 		Timestamp:      time.Now(),
+		Opaque:         entry,
 	}
 
 	// Note: confluent kafka supports the concept of one channel per connection, so we could add those here and get rid of reliableAckWorkers
 	// ex.: https://github.com/confluentinc/confluent-kafka-go/blob/master/examples/producer_custom_channel_example/producer_custom_channel_example.go#L79
-	if p.reliableAck {
-		msg.Opaque = entry
-	}
 	entry.ProduceTime = time.Now()
 	if err := p.kafkaProducer.Produce(msg, nil); err != nil {
 		p.logError(err)
 		return
 	}
 
-	metricsRegistry.produceCount.Inc(map[string]string{"record_type": entry.TxType})
-	metricsRegistry.byteTotal.Add(int64(entry.Length()), map[string]string{"record_type": entry.TxType})
```

Comment:
How about we keep these metrics and add the new ones in the producer-events handler (with different names), so we can distinguish between things we believe we produced and things that got confirmed.

Reply:
updated